### ISSUE TITLE
[SwitchMaterial] On-Off state are not vertically aligned

### ISSUE THREAD

**Issue by ftabashir** [material-components/material-components-android#1017]
**Description:** By using multiple SwitchMaterial components in multiple rows you will see they are not vertically aligned when one of them is on and another is off.

**Expected behavior:** 

I think that thumb of SwitchMaterial should stand in a position (in both on/off) that looks vertically aligned like this image (created by a photo editing app) :

![image](https://user-images.githubusercontent.com/14915828/74447266-56fd8800-4e8e-11ea-822d-87edb1428c71.png)

Current implementation of SwitchMaterial does not consider vertical alignment as you see:

![image](https://user-images.githubusercontent.com/14915828/74447206-3fbe9a80-4e8e-11ea-9cbb-7a4d0481adb4.png)

**Source code:** 
```
<?xml version="1.0" encoding="utf-8"?>
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:orientation="vertical"
    tools:context=".MainActivity">

    <com.google.android.material.switchmaterial.SwitchMaterial
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_margin="8dp"
        android:checked="true" />

    <com.google.android.material.switchmaterial.SwitchMaterial
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_margin="8dp" />

</LinearLayout>
```

**Android API version:** 29

**Material Library version:** 1.1.0

**Device:** All devices, also in design mode of android studio

**Comment by angdesign** [comment]
Hi ftabashir, this is as designed due to the thumb design being slightly larger than the underlying track. However, compared to the live examples of switches on material.io https://material.io/components/selection-controls/ the example screenshot appears to have an exaggerated offset, as well as alternate coloring. 
![comparison](https://user-images.githubusercontent.com/29785331/74490919-f581ed00-4e7e-11ea-9f43-3504150ae79c.png)

**Comment by ftabashir** [comment]
Hi @angdesign 
I did not try to exaggerate.
Second image I posted is the screenshot of using android material switch component. Source code shows the elements I used. I did not alter any attribute to increase thunb size.
Let me know If you think I did something wrong.

**Comment by hqzxzwb** [comment]
Hi @angdesign 
The example of material is good and the screenshot is real. I am facing the same problem here. So as I see this is an implementation bug, which means it is behaving the way (as in the screenshot) other than how it should behave (as in the material example).

**Comment by Mathbl** [comment]
Same issue here, 
![image](https://user-images.githubusercontent.com/906656/99432104-b125f080-28d9-11eb-9bb0-dd42c80cef01.png)

SwitchMaterial appears unaligned (tried using 1.2.1, 1.3.0-alpha02 and 1.3.0-alpha03 but every version has the same issue).

**Comment by Mathbl** [comment]
In the exemple, they also look weirdly aligned https://github.com/material-components/material-components-android/blob/master/docs/components/assets/switch/switch_example.png

Is it just me or it looks off?

**Comment by angdesign** [comment]
Following up, acknowledging the staggered effect, and we have an update in the works for Switches to align toggle position to follow the pattern below. 
![unnamed](https://user-images.githubusercontent.com/29785331/99436230-0f48d880-28c6-11eb-85ca-a26b40ce8c56.png)

**Comment by dvruijven** [comment]
Hi @angdesign 
Based on your last update I was wondering if you know any more about when this change might be implemented. 
I couldn't spot this on the first alpha for version 1.4.0, so I was wondering if this might be later added to this version or if it will be a bit later down the line for version 1.5.0 or onwards?

**Comment by drchen** [comment]
Hi I confirmed with the team and we are still waiting for the final design from our designers.

In the meantime you can workaround this by creating your own thumb and track drawables without left/right paddings - in this way the enabled and disabled state will be aligned.

You can do this as:

```
  <style name="MySwitchStyle" parent="Widget.Material3.CompoundButton.Switch">
    <item name="track">@drawable/my_track_no_paddings</item>
    <item name="android:thumb">@drawable/my_thumb_no_paddings</item>
  </style>
```

I apologize for the inconvenience here - changing design will introduce a lot of works on updating and verifying UX affected on all clients we have. Therefore we will prefer to do this at once. Thanks for your patience and understanding!

**Comment by drchen** [comment]
Hi folks, I have a good news for you. 

Our new switch design is almost finalized and I'm currently working on prototyping/design review/etc. 
The new switch is expected to be shipped with 1.7 version, as long as everything goes smoothly.

I'll close this issue since it's going to be obsolete and I think the workaround in the above comments should serve the needs for now. : ) 

Let me know if you have any concerns.
